### ISSUE TITLE
VB: Disable suggestion to simplify field names in object initializer expressions.

### ISSUE THREAD

**Issue by ericmutta** [dotnet/roslyn#23763]
**Version Used**: 

Visual Studio Community 2017 v15.5.1

**Steps to Reproduce**:

Type in the following code in Windows console project:

```vb
Public Module MainModule
  Public Class C1
    Public Text As String
    Public Number As Integer
  End Class

  Public Sub Main()
    Dim cc1 = New C1() With {.Text = "test", .Number = 12}
    Dim cc2 = New C1() With {.Text = cc1.Text, .Number = cc1.Number}
  End Sub
End Module
```

The editor seems to think that the dotted field names in the `Dim cc2...` line can be simplified. It colours them gray and when you hover over them it shows a tooltip saying so:

![vs_incorrent_can_simplify_name1](https://user-images.githubusercontent.com/20465797/33956787-b9279e52-e050-11e7-9e43-a5dc846ac06d.png)

...and:

![vs_incorrent_can_simplify_name2](https://user-images.githubusercontent.com/20465797/33956856-e4726eb6-e050-11e7-85c2-6496752832c2.png)

**Expected Behavior**:

This suggestion should not be made for VB object initialisers since they do not support inferred field names. If you actually go ahead and try this:

```vb
Dim cc2 = New C1() With {cc1.Text,  cc1.Number}
```

...the compiler complains with the error `Name of field or property being initialized in an object initializer must start with '.'` Field names _can_ be inferred for tuples so perhaps this name simplification suggestion should be implemented for VB tuples (it doesn't appear to have been implemented so far).

**Comment by jcouv** [comment]
Thanks for reporting this (especially with the details and screenshots).
It was recently reported in https://github.com/dotnet/roslyn/issues/23659 as well.

I'll close this issue a duplicate.

**Comment by ericmutta** [comment]
@jcouv many thanks for following up, glad others had seen the issue too 👍

**Comment by jcouv** [comment]
The fix should ship in 15.6 preview 3.

**Comment by ericmutta** [comment]
> @jcouv The fix should ship in 15.6 preview 3.

That was super fast! Thank's a lot for following up, very much appreciated!
